MWContentDisplayPlugin/MWContentDisplayPlugin/ContentStackStep/Local/MWContentDisplayStackViewController.swift

```diff
@@ -101,8 +101,46 @@ public class MWContentDisplayStackViewController: MWStepViewController, Workflow
             } catch {
                 self.show(error)
             }
-        } else if let linkURL = item.linkURL {
-            self.presentActivitySheet(with: [linkURL], sourceRect: rect)
+        } else if (item.linkURL != nil || item.shareText != nil || item.shareImageURL != nil) {
+            // Collect everything on a background queue (including downloading the image)
+            DispatchQueue.global(qos: .userInitiated).async {
+                var itemsToShare: [Any] = []
+                
+                if let imageURL = item.shareImageURL {
+                    do {
+                        let data = try Data(contentsOf: imageURL)
```

Comment:
@xmollv Please, use AppRail URL scheme. Not only to complete the URL (with session values). But, also to use the correct underlying Network requester.